### PR TITLE
Implement express backend

### PR DESCRIPTION
**Pull request by cstavitsky**
## Type of Change
- [ ]  Bugfix
- [x]  New feature
- [ ]  Enhancement
- [ ]  Refactoring

## Description
Implement Express (Node.js) backend. By default the flask backend is used, but you can instead connect Express backend by passing `?backend=express` as a query parameter.

#### Hitting the Express Backend
Tested the regular Empower Plant flows

- Regression test: started on homepage with no query param and clicked through to products. Saw the correct url for the default Flask backend was hit, and products were fetched successfully.

Screen Shot 2021-10-13 at 11 55 17 AM

- Started on homepage with `?backend=express`, clicked through to products, and saw the correct url for Express backend hit and products were fetched successfully.

Screen Shot 2021-10-13 at 11 52 21 AM

- Hit the products and products-join page directly with `?backend=express`, and saw that we hit the appropriate backend URLs URL (port 8088 is Express)

- Went through the checkout error flow with `?backend=express` and we can see the express backend is hit and returns the expected error

Screen Shot 2021-10-13 at 12 01 12 PM

- **Distributed tracing works**, i.e. see this [transaction event](https://sentry.io/organizations/testorg-az/discover/chris-apm-express:4a27fe4fdf234beab8b3d714fe8bf33c/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&query=transaction%3A%2A%2Fproducts-join%2A+se%3Achris&sort=-timestamp&statsPeriod=24h&yAxis=count%28%29) and [this issue](https://sentry.io/organizations/testorg-az/issues/2670583920/?project=5963130&query=is%3Aunresolved&statsPeriod=24h)

- Tags show up as they're supposed to (`release`, `se`). [Source](https://sentry.io/organizations/testorg-az/discover/chris-apm-express:4a27fe4fdf234beab8b3d714fe8bf33c/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&query=transaction%3A%2A%2Fproducts-join%2A+se%3Achris&sort=-timestamp&statsPeriod=24h&yAxis=count%28%29)

Screen Shot 2021-10-13 at 12 07 30 PM

## Links to Sentry issues

- [`/products` performance](https://sentry.io/organizations/testorg-az/discover/chris-apm-express:953658d5a3d54b72a050a28c743ec79f/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&query=transaction%3A%2A%2Fproducts%2A+se%3Achris&sort=-timestamp&statsPeriod=24h&yAxis=count%28%29) - Frontend application-monitoring-javascript, backend my test Express project
- [`/products-join` performance](https://sentry.io/organizations/testorg-az/discover/chris-apm-express:4a27fe4fdf234beab8b3d714fe8bf33c/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&query=transaction%3A%2A%2Fproducts-join%2A+se%3Achris&sort=-timestamp&statsPeriod=24h&yAxis=count%28%29) - Frontend application-monitoring-javascript, backend my test Express project
-  [Error caused by checkout](https://sentry.io/organizations/testorg-az/issues/2670583920/?project=5963130&query=is%3Aunresolved&statsPeriod=24h)

### CODE REVIEW

**Review by thinkocapo — CHANGES_REQUESTED** [review]
Can you run
```
rm -rf express/node_modules
echo 'express/node_modules' >> .gitignore
## or 
echo '**/node_modules' >> .gitignore
```
basically delete express/node_modules, then commit this change, and reflect in gitignore that we want to ignore node_modules. (if you forget tgitignore then it will commit node_modules all over again)

and then push your changes to github, and this should remove the 400k committed node_modules from the github repo.

then I can review. right now I have to scroll past 4,000 changed files, the website started freezing on me.

**Comment by cstavitsky** [comment]
@thinkocapo yep, I'll go ahead and take care of that. My bad - assumed that everything in `node_modules/` were files we'd want to commit.

File: express/server.js [hunk 1]
```diff
@@ -14,19 +14,107 @@
...
+    Sentry.captureException(error);
+    res.status(500).send(error);
+  }
+
+});
+
 // Listen to the App Engine-specified port, or 8080 otherwise
 const PORT = process.env.PORT || 8080;
 app.listen(PORT, () => {
   console.log(`Server listening on port ${PORT}...`);
```
Comment (cstavitsky) [inline express/server.js:116]: Note to self: update this to default to port `8081`!

File: .gitignore [hunk 2]
```diff
@@ -37,4 +38,5 @@ node/package-lock.json
 ruby/Gemfile.lock
 flask/test.py
 react/app.yaml
-flask/app.yaml
\ No newline at end of file
+flask/app.yaml
```
Comment (cstavitsky) [inline .gitignore:41]: This could also just be `**/app.yaml` if we expect to add more backend other than express and springboot 🤷

**Review by thinkocapo — APPROVED** [review]

**Pull request merged**